3-0-spring-framework/3-3-0-enable-string-trimming/src/test/java/com/bobocode/TrimmingPostProcessorTest.java

```diff
@@ -0,0 +1,254 @@
+            "config class by @EnableStringTrimming")
+    void stringTrimmingConfiguration_annotatedAsConfiguration() {
+        Optional<Class<?>> configClass = findClassInPackage("StringTrimmingConfiguration");
+        assertTrue(configClass.isPresent());
+        assertNull(configClass.get().getDeclaredAnnotation(Configuration.class));
+    }
+
+    @Test
+    @Order(7)
+    @DisplayName("StringTrimmingConfiguration contains methods to provide new instance of TrimmedAnnotationBeanPostProcessor")
+    void stringTrimmingConfiguration_createsBeanOfTrimmedAnnotationPostProcessor() {
+        Optional<Class<?>> configClass = findClassInPackage("StringTrimmingConfiguration");
+
+        Method[] methods =
+                configClass.orElseThrow(() -> new NoSuchElementException("StringTrimmingConfiguration class is not found"))
+                        .getDeclaredMethods();
+
+        Optional<Method> postProcessorBeanMethod =
+                Stream.of(methods).filter(m -> m.getReturnType().equals(TrimmedAnnotationBeanPostProcessor.class))
+                        .findFirst();
+        assertTrue(postProcessorBeanMethod.isPresent());
+    }
+
+    @Test
+    @Order(8)
+    @DisplayName("Methods of StringTrimmingConfiguration that provides TrimmedAnnotationBeanPostProcessor instance annotated with" +
+            " @Bean annotation")
+    void trimmedAnnotationBeanPostProcessorMethod_markedAsBean() {
+        Optional<Class<?>> configClass = findClassInPackage("StringTrimmingConfiguration");
+        Method[] methods = configClass.orElseThrow(
+                () -> new NoSuchElementException("StringTrimmingConfiguration class is not found")).getDeclaredMethods();
+        Optional<Bean> beanAnnotation = Stream.of(methods)
+                .filter(m -> m.getReturnType().equals(TrimmedAnnotationBeanPostProcessor.class))
+                .map(m -> m.getAnnotation(Bean.class)).findFirst();
+        assertTrue(beanAnnotation.isPresent());
+    }
+
+    @Test
+    @Order(9)
+    @DisplayName("TrimmedAnnotationBeanPostProcessor implements BeanPostProcessor interface")
+    void trimmedAnnotationBeanPostProcessor_implementsBeanPostProcessor() {
+        boolean isBeanPostProcessorPresent = Arrays.stream(TrimmedAnnotationBeanPostProcessor.class.getInterfaces())
+                .anyMatch(i -> i.isAssignableFrom(
+                        BeanPostProcessor.class));
+        assertTrue(isBeanPostProcessorPresent);
+    }
+
+    @Test
+    @Order(10)
+    @DisplayName("TrimmedAnnotationBeanPostProcessor overrides postProcessAfterInitialization() method")
+    void trimmedAnnotationBeanPostProcessor_overridesPostProcessAfterInitMethod() {
+        List<String> methodNames = Arrays.stream(TrimmedAnnotationBeanPostProcessor.class.getDeclaredMethods())
+                .map(Method::getName)
+                .toList();
+        assertThat(methodNames, contains("postProcessAfterInitialization"));
```

Comment:
@Stanislav-Zabramnyi please use **AssertJ instead of Hamcrest.** 

For instance, 
* `assertThat(methodNames, contains("postProcessAfterInitialization"));`
VS.
* `assertThat(methodNames).contains("postProcessAfterInitialization");`